### ISSUE TITLE
False positive "unnecessary quotes" warning

### ISSUE THREAD

**Issue by fertapric** [rrrene/credo#569]
### Environment

* Credo version (`mix credo -v`): 0.9.3
* Erlang/Elixir version (`elixir -v`):
    
    ```
    Erlang/OTP 21 [erts-10.0.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

    Elixir 1.7.0 (compiled with Erlang/OTP 21)
    ```
* Operating system: macOS High Sierra

### What were you trying to do?
 
Running `mix credo --strict` produces warnings that do not appear when compiling the code.

Here is an example:

```elixir
defmodule Foo do
  def foo do
    with {:ok, a} <- echo("a"),
         {:ok, b} <- echo(:"b_#{a}_") do
      "#{a} #{b}"
    end
  end

  defp echo(value) do
    {:ok, value}
  end
end
```

The current code produces:

```
warning: found quoted atom "b__" but the quotes are not required. Quotes should only be used to introduce atoms with foreign characters in them
  nofile:4
```

### Expected outcome

No warnings for the given use case.

**Comment by rrrene** [comment]
@fertapric Thx for reporting! :+1: Could you try Credo `v0.10.0` and report back if it solves this problem?

**Comment by fertapric** [comment]
@rrrene Unfortunately, v0.10.0 does not solve this problem. 

Proof [in this PR](https://github.com/fertapric/async_with/pull/24). The warnings in the [build](https://travis-ci.org/fertapric/async_with/jobs/409690658) are related to [this line](https://github.com/fertapric/async_with/blob/master/test/async_with_test.exs#L843), but that line does not emit warnings when is compiled by Elixir.

**Comment by rrrene** [comment]
Found it! 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by fertapric** [comment]
It worked! 👏
